### ISSUE TITLE
Run intrinsic doesn't work on Windows because of Linux specific calls

### ISSUE THREAD

**Issue by Jackojc** [wotpp/wotpp#29]
As the title says.

**Comment by jlagarespo** [comment]
Wouldn't calls like `popen` already work with a posix-compliant toolchain such as mingw?

**Comment by Jackojc** [comment]
I don't think MinGW translates calls to popen.

**Comment by jlagarespo** [comment]
I do think cygwin has though http://www.polarhome.com/service/man/?qf=popen&tf=2&of=Cygwin&sf=

**Comment by Jackojc** [comment]
I mean I'd rather get it to work without having to jump through those hoops.

**Comment by jlagarespo** [comment]
Yeah well, since the STL doesn't support processes natively, I guess our only option is to abstract them away and hide the platform-dependent code inside some ifdefs. But I'm not sure how much easier can that be than compiling it using a posix-compliant toolchain.

**Comment by Jackojc** [comment]
I don't think it's too hard to work with, it's less than ideal sure but it's the best option we have.
